### PR TITLE
gROOTMutex -> RWLock

### PR DESCRIPTION
**Pull request by pcanal**
One pending question (beside whether this is performant enough) is whether to keep the old TRWSpinLock and the new TRWSpinLock (Reentrant) or to have them (as in this MR) the same.

### CODE REVIEW

**Comment by pcanal** [comment]
@etejedor Could you take a look?

File: core/base/inc/TPluginManager.h [hunk 1]
```diff
@@ -155,7 +155,7 @@ friend class TPluginManager;
       // resource ... and both SetParams and Execute ends up taking the lock
       // individually anyway ...
 
-      R__LOCKGUARD2(gInterpreterMutex);
+      R__LOCKGUARD(gInterpreterMutex);
```
Comment (etejedor) [inline core/base/inc/TPluginManager.h]: Should we also apply the same fix to all occurrences of R__LOCKGUARD_IMT2 and replace them by R__LOCKGUARD_IMT?

Comment (pcanal) [inline core/base/inc/TPluginManager.h]: It depends on whether the mutex guard is guaranteed to be initialized or not ....

**Comment by dpiparo** [comment]
If my two minor comments are addressed this pr for me is ready to go.

**Comment by pcanal** [comment]
@dpiparo I don't see the comments.

Besides, after further development, I found the need also make the lock re-entrant from Read to Write lock making it relatively heavy so I am planning to actually separate the simple RW lock from the re-entrant one.

So once I find (and address) the comments, I will close this.

**Comment by pcanal** [comment]
@phsft-bot build please.

**Pull request closed without merging**